sae_lens/sae_training_runner.py

```diff
@@ -187,58 +183,57 @@ def _init_sae_group_b_decs(
     def save_checkpoint(
         self,
         trainer: SAETrainer,
-        checkpoint_name: int | str,
+        checkpoint_name: str,
         wandb_aliases: list[str] | None = None,
-    ) -> str:
-
-        checkpoint_path = f"{trainer.cfg.checkpoint_path}/{checkpoint_name}"
+    ) -> None:
+        """Save a checkpoint of the SAE locally and optionally to wandb."""
 
-        os.makedirs(checkpoint_path, exist_ok=True)
+        base_path = Path(self.cfg.checkpoint_path) / checkpoint_name
+        base_path.mkdir(exist_ok=True)
 
-        path = f"{checkpoint_path}"
-        os.makedirs(path, exist_ok=True)
+        self.activations_store.save_state(base_path)
 
+        # TODO(oli-clive-griffin): Is this broken? it seems like this is "assymetrical" in the sense that
+        # this changes the output of the SAE.
+        # Should we not also balance this by scaling the encoder weights?
         if self.sae.cfg.normalize_sae_decoder:
             self.sae.set_decoder_norm_to_unit_norm()
-        self.sae.save_model(path)
+
+        weights_path, cfg_path, sparsity_path = self.sae.save_model(
+            str(base_path),
+            trainer.log_feature_sparsity,
+        )
 
         # let's over write the cfg file with the trainer cfg, which is a super set of the original cfg.
         # and should not cause issues but give us more info about SAEs we trained in SAE Lens.
-        config = trainer.cfg.to_dict()
-        with open(f"{path}/cfg.json", "w") as f:
+        config = self.cfg.to_dict()
+        with open(cfg_path, "w") as f:
             json.dump(config, f)
 
-        log_feature_sparsities = {"sparsity": trainer.log_feature_sparsity}
-
-        log_feature_sparsity_path = f"{path}/{SPARSITY_PATH}"
-        save_file(log_feature_sparsities, log_feature_sparsity_path)
```

Comment:
`self.sae.save_model` already does this if you pass the sparsity tensor in. we can remove duplicate/coupled code this way